crates/pop-common/src/templates/frontend.rs

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use strum::{EnumProperty as _, VariantArray};
+use strum_macros::{AsRefStr, Display, EnumMessage, EnumProperty, EnumString};
+
+use crate::templates::{Template, Type};
+
+/// Supported frontend template types.
+#[derive(
+	AsRefStr, Clone, Default, Debug, Display, EnumMessage, EnumString, Eq, PartialEq, VariantArray,
+)]
+pub enum FrontendType {
+	/// Contract-based frontend templates.
+	#[default]
+	#[strum(ascii_case_insensitive, serialize = "contract", message = "Contract")]
+	Contract,
+	/// Chain-based frontend templates.
+	#[strum(ascii_case_insensitive, serialize = "chain", message = "Chain")]
+	Chain,
+}
+impl Type<FrontendTemplate> for FrontendType {
+	fn default_template(&self) -> Option<FrontendTemplate> {
+		match &self {
+			FrontendType::Contract => Some(FrontendTemplate::Typink),
+			FrontendType::Chain => Some(FrontendTemplate::CreateDotApp),
+		}
+	}
+}
+
+/// Supported frontend templates.
+#[derive(
+	AsRefStr,
+	Clone,
+	Debug,
+	Default,
+	Display,
+	EnumMessage,
+	EnumProperty,
+	EnumString,
+	Eq,
+	Hash,
+	PartialEq,
+	VariantArray,
+)]
+pub enum FrontendTemplate {
+	/// Typeink template: The ultimate toolkit for dApps development on Polkadot, powered by <https://dedot.dev>!.
+	#[default]
+	#[strum(
+		serialize = "typink",
+		message = "Typink",
+		detailed_message = "The ultimate toolkit for dApps development on Polkadot, powered by https://dedot.dev",
+		props(Command = "create-typink@latest", Type = "Contract",)
+	)]
+	Typink,
+	/// Inkathon template: Next generation full-stack boilerplate for ink! smart contracts running
+	/// on PolkaVM.
+	#[strum(
+		serialize = "inkathon",
+		message = "Inkathon",
+		detailed_message = "Next generation full-stack boilerplate for ink! smart contracts running on PolkaVM.",
+		props(Command = "create-inkathon-app@latest", Type = "Contract",)
+	)]
+	Inkathon,
+	/// Create Dot App template: A command-line interface (CLI) tool designed to streamline the
+	/// development process for Polkadot-based decentralized applications (dApps).
+	#[strum(
+		serialize = "create-dot-app",
+		message = "create-dot-app",
+		detailed_message = "A command-line interface (CLI) tool designed to streamline the development process for Polkadot-based decentralized applications (dApps)",
+		props(Command = "create-dot-app@latest", Type = "Chain",)
+	)]
+	CreateDotApp,
+}
+
+impl Template for FrontendTemplate {}
+
+impl FrontendTemplate {
+	/// Get the command to create a new project using this template.
+	pub fn command(&self) -> Option<&str> {
+		self.get_str("Command")
```

Comment:
Can't we `unwrap()` here?

Reply:
We have been trying to avoid unwrap(), once getting the command in the pop-cli we throw an error if no command has been configured instead:
```
let command = template
		.command()
		.ok_or_else(|| anyhow::anyhow!("no command configured for {:?}", template))?;
```

Reply:
My point is that we can be 100% `.unwrap()` won't fail there. Am I correct? I mean, all variants of the enum have the property `Command` defined, so in principle there is no room for failure.